sdk/aqueduct/decorator.py

```diff
@@ -373,15 +407,25 @@ def op(
 
         >>> recommendations.get()
     """
-    _type_check_decorator_arguments(
-        description,
-        file_dependencies,
-        requirements,
+    # Establish parity between `num_outputs` and `outputs`, or raise exception if there is a mismatch.
+    if num_outputs is None and outputs is None:
+        num_outputs = 1
+    elif num_outputs is not None and outputs is not None:
+        if len(outputs) != num_outputs:
+            raise InvalidUserArgumentException(
+                "`len(outputs) must be equivalent to `num_outputs`. Getting %d and %d"
+                % (len(outputs), num_outputs)
+            )
+    elif num_outputs is None and outputs is not None:
+        num_outputs = len(outputs)
+
+    # If not set, default number of outputs is one.
```

Comment:
I'm a bit confused as after the above 3 if-elif blocks the only remaining condition is `num_output is not None` and outputs is None` ? It seems that we should do some assertion here instead?

Reply:
Yeah the casework here is a little tricky - if a `num_output` is provided, then the `outputs` is an optional argument, hence why that case isn't specified here. Open to suggestions, but the relationship isn't symmetric